homework3/irina.butakova/BeanPostProcessorForLocalization/src/main/java/LocalizedValueBeanPostProcessor.java

```
@@ -0,0 +1,56 @@
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+public class LocalizedValueBeanPostProcessor implements BeanPostProcessor {
+
+    public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException{
+        return bean;
+    }
+
+    public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
+        Arrays
+                .stream(bean.getClass().getDeclaredFields())
+                .filter(this::isStringField)
+                .filter(this::isAnnotatedWithLocalized)
+                .forEach(field -> getKeyAndSetValue(bean, field));
+
+
+        return bean;
+    }
+
+    private boolean isStringField(Field field) {
+        return field.getType().equals(String.class);
+    }
+
+    private boolean isAnnotatedWithLocalized(Field field) {
+        return field.isAnnotationPresent(Localized.class);
+    }
+
+    private void getKeyAndSetValue(Object bean, Field field) {
+        String key = getKey(bean, field);
+        setValue(bean, field, getValue(key));
+    }
+
+    private String getKey(Object bean, Field field) {
+        return field.getAnnotation(Localized.class).key();
+    }
+
+    private String getValue(String key) {
+        return ResourceBundle.getBundle("locale/navigation", Locale.getDefault()).getString(key);
+    }
+
+    private void setValue(Object bean, Field field, String value) {
+        field.setAccessible(true);
+        try {
+            field.set(bean, value);
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
```

Think how to avoid hiding exceptions